Source/QRCodeModel.swift

```diff
@@ -251,28 +251,23 @@
             var offset = 0
             var maxDcCount = 0
             var maxEcCount = 0
-            var dcdata = [[Int]!](repeating: nil, count: rsBlocks.count)
-            var ecdata = [[Int]!](repeating: nil, count: rsBlocks.count)
-            for r in rsBlocks.indices {
-                let dcCount = rsBlocks[r].dataCount
-                let ecCount = rsBlocks[r].totalCount - dcCount
+            var dcdata = [[Int]]()
+            var ecdata = [[Int]]()
```

Comment:
```suggestion
            dcdata.reserveCapacity(rsBlocks.count)
            var ecdata = [[Int]]()
            ecdata.reserveCapacity(rsBlocks.count)
```

Since we know the sizes of the arrays, it's probably? better to call reserve capacity before using them